internal/validation/mock_trace_validation.go

```diff
@@ -77,21 +80,31 @@ func AddSpans(uploadedSpans map[string]*cloudtrace.Span, lock *sync.Mutex,
 		lock.Lock()
 		defer lock.Unlock()
 		for _, span := range spans {
-			if _, ok := uploadedSpans[span.Name]; ok {
+			if _, ok := uploadedSpanNames[span.Name]; ok {
 				br.Reason = span.Name
 				st, err := ErrDuplicateSpanName.WithDetails(br)
 				if err != nil {
 					panic(fmt.Sprintf("unexpected error attaching metadata: %v", err))
 				}
 				return st.Err()
 			}
-			uploadedSpans[span.Name] = span
+			*uploadedSpans = append(*uploadedSpans, span)
+			uploadedSpanNames[span.Name] = true
 		}
 	}
 
 	return nil
 }
 
+// AccessSpan returns the span at the given index if it is in range.
+// If it is not in range, an error is returned.
+func AccessSpan(index int, uploadedSpans []*cloudtrace.Span) (*cloudtrace.Span, error) {
```

Comment:
Don't forget to lock/unlock when accessing the uploadedSpan

Reply:
Nevermind, I saw the other lock.

Reply:
Yeah I lock/unlock in the function that calls `AccessSpan`